react-components/src/components/Widgets/WindowWidget.tsx

```diff
@@ -20,31 +20,35 @@ import {
 } from './constants';
 import { useTranslation } from '../i18n/I18n';
 import { withSuppressRevealEvents } from '../../higher-order-components/withSuppressRevealEvents';
+import { StyledComponent, WidgetBody, WidgetContent } from './elements';
 
 type WindowWidgetProps = {
   title?: string;
   subtitle?: string;
   header?: string;
   type?: string;
   children: ReactNode;
+  onClose?: () => void;
```

Comment:
Could this handle both open-and close-events? It could e.g. take a boolean as argument to indicate whether it was opened or closed.